pandas/tools/pivot.py

```diff
@@ -434,12 +437,44 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    # Sanity checks
+    if values is None and callable(aggfunc):
```

Comment:
In [2]: df = pd.DataFrame({'a': [1, 2, 2, 2, 2], 'b': [3, 3, 4, 4, 4]})
    In [3]: pd.crosstab(df.a, df.b, aggfunc=np.mean) 
    Out[3]: 
    b  3  4
    a      
    1  1  0
    2  1  3